### PR TITLE
add page for behavioral targeting

### PR DESCRIPTION
**Pull request by jeddai**
## Description (optional)

This PR adds a page explaining behavioral targeting for general purposes, writing experiments, and engineering.

### CODE REVIEW

File: docs/deep-dives/mobile/behavioral-targeting.mdx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+| `eventAveragePerNonZeroInterval` | Calculates the average of all buckets with a non-zero value within the range | `interval, bucket_count, starting_bucket` | float |
+| `eventLastSeen` | Returns the number of whole time intervals between the starting bucket and the first bucket with a non-zero value | `interval, starting_bucket` | int |
+
+## Designing Experiments
+
+The following are the existing options for behavioral targeting as defined in Experimenter.
+
+| Name | Description | Targeting String |
+|:---|:---|:---|
+| Core Active Users | A user who has used the application 21 out of the last 28 days. | <pre><code>'<event\>'&#124;eventCountNonZero('Days', 28, 0) >= 21</code></pre> |
```
Comment (travis79) [inline docs/deep-dives/mobile/behavioral-targeting.mdx:59]: nit: `'<event\>'` here is just a little confusing, it took me a second to understand that it wasn't something that didn't render properly when looking at this as rendered markdown.  Maybe be specific and just call it `activity-event` here, and `sign-in-event` or something generic would have helped. This is just a nit, feel free to disregard, I _did_ get the meaning once I looked at it a little harder :)

**Review by travis79 — APPROVED** [review]
🤩

File: docs/deep-dives/mobile/behavioral-targeting.mdx (hunk 1 above)
Comment (jeddai) [reply docs/deep-dives/mobile/behavioral-targeting.mdx:59]: Yeah if I put it without the `\` the page rendering failed. I'll adjust it, I think that's reasonable!

File: docs/deep-dives/mobile/behavioral-targeting.mdx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+## Designing Experiments
+
+The following are the existing options for behavioral targeting as defined in Experimenter.
+
+| Name | Description | Targeting String |
+|:---|:---|:---|
+| Core Active Users | A user who has used the application 21 out of the last 28 days. | <pre><code>'app-opened-event'&#124;eventCountNonZero('Days', 28, 0) >= 21</code></pre> |
+| Recently Logged In Users | A user who has logged into Sync within the last 12 weeks. | <pre><code>'sync-signin-event'&#124;eventCountNonZero('Weeks', 12, 0) > 0</code></pre> |
+
```
Comment (jhugman) [inline docs/deep-dives/mobile/behavioral-targeting.mdx:61]: Could you add some more examples to help build some intuition around how to use this e.g. is this the first time the app has been launched today/this week? Has the app been launched in the last three days?

File: docs/deep-dives/mobile/behavioral-targeting.mdx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+## Engineering
+
+In order to instrument a new behavior/event, an equivalent call to the Nimbus event recording method must be made alongside the call to record a Glean event.
```
Comment (jhugman) [inline docs/deep-dives/mobile/behavioral-targeting.mdx:65]: Worth adding a sidebar to say that this is a temporary arrangement, until Glean allow Nimbus to mirror Glean events.

File: docs/deep-dives/mobile/behavioral-targeting.mdx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+[Example in PR](https://github.com/mozilla-mobile/fenix/pull/27934)
+
+</TabItem>
+</Tabs>
+
+<hr/>
+
+The following are the events that are currently instrumented in Firefox for iOS and Android, respectively:
```
Comment (jhugman) [inline docs/deep-dives/mobile/behavioral-targeting.mdx:131]: Nit: I'd put this above the engineering section, and give it a heading— so we can link to it.

**Review by jhugman — APPROVED** [review]
r+wc. Thanks.

**Pull request merged**